src/core/qgis.h

```diff
@@ -555,25 +555,31 @@ typedef unsigned long long qgssize;
 #define NODISCARD [[nodiscard]]
 #elif defined(_MSC_VER)
 #define NODISCARD // no support
-#elif __has_cpp_attribute(nodiscard)
-#define NODISCARD [[nodiscard]]
-#elif __has_cpp_attribute(gnu::warn_unused_result)
+#elif defined(__has_cpp_attribute)
+#if __has_cpp_attribute(gnu::warn_unused_result)
 #define NODISCARD [[gnu::warn_unused_result]]
 #else
 #define NODISCARD Q_REQUIRED_RESULT
```

Comment:
I would prefer to keep the native [[nodiscard]] rather than Qt implementation. So basically do not remove the lines 558 and 559.